### PR TITLE
Update hist() docstring following removal of normed kwarg.

### PR DESCRIPTION
**Pull request by anntzer**
and change the default of density from None to False.  (This doesn't
change anything from the API PoV as we convert the kwarg using `bool`
anyways, but the default was previously `None` for the sole purpose of
detecting when both `normed` and `density` were both passed and error
out in that case.)

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
very, very slowly :p

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -6423,12 +6419,15 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
...
-            last bin equals 1. If *cumulative* evaluates to less than 0
-            (e.g., -1), the direction of accumulation is reversed.
-            In this case, if *normed* and/or *density* is also ``True``, then
-            the histogram is normalized such that the first bin equals 1.
+            gives the total number of datapoints.
+
+            If *density* is also ``True`` then the histogram is normalized such
+            that the last bin equals 1.
+
+            If *cumulative* is to less than 0 (e.g., -1), the direction of
```
Comment (timhoffm) [inline lib/matplotlib/axes/_axes.py]:
```suggestion
            If *cumulative* is a number less than 0 (e.g., -1), the direction of
```
That should be mentioned in the type. Just `bool` is plain confusing. Maybe we could also deprecate numbers in favor of "reversed", but that's for another PR.

Comment (anntzer) [reply lib/matplotlib/axes/_axes.py]: I now documented the type as `bool or -1` which is technically too restrictive but more informative in practice, I think.

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.1.x

**Comment by tacaswell** [comment]
This is documenting changes in https://github.com/matplotlib/matplotlib/pull/14244 which is targetting 3.2.

**Pull request merged**